### PR TITLE
app-benchmarks/sysbench: bump to 1.0.12

### PR DESCRIPTION
**Pull request by hydrapolic**
Package-Manager: Portage-2.3.18, Repoman-2.3.6

### CODE REVIEW

File: app-benchmarks/sysbench/sysbench-1.0.11.ebuild [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	sed -i -e "/^htmldir =/s:=.*:=/usr/share/doc/${PF}/html:" doc/Makefile.am || die
+
+	# remove bundled libs
+	rm -r third_party/luajit/luajit third_party/concurrency_kit/ck third_party/cram || die
+
+	./autogen.sh || die
+}
+
+src_configure() {
+	local myeconfargs=(
```
Comment (titanofold) [inline app-benchmarks/sysbench/sysbench-1.0.11.ebuild]: Why can't we just pass these arguments directly to econf?

Comment (hydrapolic) [reply app-benchmarks/sysbench/sysbench-1.0.11.ebuild]: @titanofold, yes we could, but each lines must be then backslashed which looks a bit uglier I suppose. This approach seems like a standard for most packages, so I would keep it like this.

File: app-benchmarks/sysbench/sysbench-1.0.11.ebuild [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	dev-libs/concurrencykit
+	dev-libs/libxslt
+	sys-devel/libtool:=
+	virtual/pkgconfig
+	test? ( dev-util/cram )"
+
+src_prepare() {
+	default
+
+	sed -i -e "/^htmldir =/s:=.*:=/usr/share/doc/${PF}/html:" doc/Makefile.am || die
```
Comment (SoapGentoo) [inline app-benchmarks/sysbench/sysbench-1.0.11.ebuild]: please fix this properly with a patch, which can then be sent upstream

File: app-benchmarks/sysbench/sysbench-1.0.11.ebuild [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+	# remove bundled libs
+	rm -r third_party/luajit/luajit third_party/concurrency_kit/ck third_party/cram || die
+
+	./autogen.sh || die
```
Comment (SoapGentoo) [inline app-benchmarks/sysbench/sysbench-1.0.11.ebuild]: what does this do? are you autoreconf'ing without depending on `autotools.eclass`? Try to never use `autogen.sh` scripts, and instead just run vanilla `eautoreconf`

File: app-benchmarks/sysbench/sysbench-1.0.11.ebuild [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		--with-system-ck
+	)
+
+	econf "${myeconfargs[@]}"
+}
+
+src_compile() {
+	default
+
+	pax-mark m "${S}/src/${PN}"
```
Comment (SoapGentoo) [inline app-benchmarks/sysbench/sysbench-1.0.11.ebuild]: shouldn't this rather be done in the install tree in `src_install`?

**Comment by hydrapolic** [comment]
Thank you @SoapGentoo for the suggestions, pushed an updated version.

File: app-benchmarks/sysbench/sysbench-1.0.12.ebuild [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+src_install() {
+	default
+
+	pax-mark m "${D}/usr/bin/${PN}"
+}
+
+src_test() {
+	emake check test
+}
```
Comment (SoapGentoo) [inline app-benchmarks/sysbench/sysbench-1.0.12.ebuild]: 1. needs to go before `src_install`
2. Having a look at https://github.com/akopytov/sysbench/blob/master/tests/Makefile.am seems to suggest that `make test` runs the exact same stuff as `make check`, hence try obliterating `src_test` completely and see whether the defaults work

File: app-benchmarks/sysbench/sysbench-1.0.12.ebuild [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		--with-system-ck
+	)
+
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	default
+
+	pax-mark m "${D}/usr/bin/${PN}"
```
Comment (SoapGentoo) [inline app-benchmarks/sysbench/sysbench-1.0.12.ebuild]: `pax-mark m "${ED}"/usr/bin/${PN}` (dont unnecessarily break Prefix)

File: app-benchmarks/sysbench/sysbench-1.0.12.ebuild (hunk 5 above)
Comment (hydrapolic) [reply app-benchmarks/sysbench/sysbench-1.0.12.ebuild]: Ad 2: check does not run the tests, it's just a check on the actual tests, the test target runs the tests.

**Comment by hydrapolic** [comment]
Thanks @SoapGentoo for checking this again, thanks for your time. Pushed the updated version, tests passing on ~amd64.

File: app-benchmarks/sysbench/sysbench-1.0.12.ebuild [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+	# remove bundled libs
+	rm -r third_party/luajit/luajit third_party/concurrency_kit/ck third_party/cram || die
+
+	eautoreconf
+}
+
+src_configure() {
+	local myeconfargs=(
+		$(use_enable aio aio)
```
Comment (mgorny) [inline app-benchmarks/sysbench/sysbench-1.0.12.ebuild]: Nit: $2 defaults to $1.

File: app-benchmarks/sysbench/sysbench-1.0.12.ebuild [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+src_test() {
+	emake check test
+}
+
+src_install() {
+	default
+
+	pax-mark m "${ED}"/usr/bin/${PN}
```
Comment (mgorny) [inline app-benchmarks/sysbench/sysbench-1.0.12.ebuild]: Nit: ED contains trailing slash in EAPI 6 and older, we usually do `${ED%/}`.

**Comment by hydrapolic** [comment]
Thank you @mgorny

**Pull request closed without merging**